### ISSUE TITLE
Timeout causes server to crash

### ISSUE THREAD

**Issue by kyc3** [mrothenbuecher/zpl-rest#11]
Hey @mrothenbuecher,

first of all, awesome work here! Helped me a lot in my project :)

By accident I found a little issue when my printer was not reachable. I called the `/rest/print` endpoint, which correctly responded with `failure: true`. After a little more time, the server then crashed with these logs

```
Mon May 01 2023 13:46:05 GMT+0200 (Central European Summer Time) Error: connect ETIMEDOUT 192.168.1.22:9100
node:_http_outgoing:644
    throw new ERR_HTTP_HEADERS_SENT('set');
    ^

Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client

```

I looked into it and nailed it down to the `client.on('error', ...)` handler, which was called a long time after the timeout handler was called. The callback then calls `res.send` which causes this error because the response was already sent.
However, I fixed it locally by making this small change [here](https://github.com/mrothenbuecher/zpl-rest/blob/master/main.js#L328-L333):

```
client.setTimeout(5000, function() {
    console.error((new Date()) + " " + "connection timed out");
    job.failed = true;
    job.error = "connection timed out";
    callback(job);
    client.destroy(); // was client.end() previously
  });
```

Basically by destroying the client, it is ensured that the other callbacks are not called anymore. Do you want me to submit a pull request?

**Comment by mrothenbuecher** [comment]
Hey,

of course you can! Thx for your work!

**Comment by kyc3** [comment]
There you go @mrothenbuecher: #12
